chain/network/src/types.rs

```diff
@@ -327,6 +327,21 @@ pub struct FullPeerInfo {
     pub partial_edge_info: PartialEdgeInfo,
 }
 
+impl From<&FullPeerInfo> for PeerInfoView {
+    fn from(full_peer_info: &FullPeerInfo) -> Self {
+        PeerInfoView {
+            addr: match full_peer_info.peer_info.addr {
+                Some(socket_addr) => socket_addr.to_string(),
+                None => "N/A".to_string(),
```

Comment:
or maybe we can make 'addr' an Option field instead?

Reply:
How it’s shown in the user interface is separate matter. You can have `foo.addr || 'N/A'`.

Reply:
The other thing is, this `addr` field needs to be deepsize-able. That is the other reason why a string is preferred here.

Reply:
I tried deleting the deepsize thing and it worked. So now I am able to change this one as well.

Reply:
See https://buildkite.com/nearprotocol/nearcore/builds/14403#be1e23a6-82fa-4c13-a569-48ff56616a36. Deepsize is required on `NetworkInfoView` and `PeerInfoView`. However, `SocketAddr` is not deep-sizeable, and must be replaced by an alternative type.

Since this struct is a view and is only responsible for storing the presentation of the data, rather than the data itself, a string should be okay.

Reply:
You could get around this problem by implementing the deepsize trait manually. Example: https://github.com/near/nearcore/blob/master/chain/network-primitives/src/types.rs#L187-L191